src/interaction.ts

```diff
@@ -139,6 +141,7 @@ export default class Zoom {
         } catch (error) {}
         this.transform = event.transform;
         this.restart_timer(10 * 1000);
+        this._consumer_on_zoom_callback?.(event.transform);
```

Comment:
Instead of looking in `_consumer_on_zoom_callback`, just pull the function straight from a new optional function stashed at`this.scatterplot.on_zoom`.